synthetics/test_schema.go

```diff
@@ -166,133 +165,133 @@ func makeTestIPSchema(mode schemaMode) *schema.Schema {
 }
 
 func makeTestAgentSchema(mode schemaMode) *schema.Schema {
-	return makeNestedObjectSchema(mode, map[string]*schema.Schema{
+	return makeNestedObjectSchemaOptional(mode, map[string]*schema.Schema{
 		"target": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 		},
 	})
 }
 
 func makeTestFlowSchema(mode schemaMode) *schema.Schema {
-	return makeNestedObjectSchema(mode, map[string]*schema.Schema{
+	return makeNestedObjectSchemaOptional(mode, map[string]*schema.Schema{
 		"target": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
```

Comment:
What is the difference between:
```
Required: requiredOnCreate(mode)
```
and
```
Optional: true
```
? The `target` attribute is required on create in pretty much everywhere. So, why have it as `Required: requiredOnCreate(mode)` for `agent` type test and as `Optional: True` for `flow` test type?

Reply:
`Required: requiredOnCreate(mode)` attribute must be passed by user on create. `Optional: true` can be optionally be passed by user on create.
This seems indeed inconsistent. I'll check that.